### DISCUSSION TITLE
Leaf Docs Total "And" Panels

### DISCUSSION THREAD

**Discussion by treeverti** [uwrit/leaf#609] [Q&A] [answered]
When I was exploring leaf docs, it appears that the ui does not allow more than three "and" boxes. I was wondering if there was a way to change that.

**Comment by ndobb** [comment, marked as answer]
Hey @treeverti! Thanks for your patience.

Yes, there are only three panels allowed at a time. This is by design, though I recognize it may be confusing or not ideal for other folks.

The reason: just from experience, I think users actually find the limitless addition of panels (e.g., users could go ahead and add 10 panels if they wanted) visually very confusing. I get that my opinion is of course subjective, but my intuition has always been that more panels goes against Leaf's overarching goal of managing complexity and keeping the UI clean.

So what can we do instead? 

Ideally users can break more complex queries into smaller, reusable building blocks. Each Saved Cohort become their own query, encapsulating saved logic while keeping things visually more understandable (or so I would hope).

![example](https://github.com/user-attachments/assets/afe8f740-e1d0-4a1e-9fe3-95f6aad00301)

Hope this helps!
